src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabProjectSCMEvent.java

```diff
@@ -54,7 +68,60 @@ public String getSourceName() {
 
     @Override
     public boolean isMatch(@NonNull SCMNavigator navigator) {
-        return navigator instanceof GitLabSCMNavigator && ((GitLabSCMNavigator) navigator).getNavigatorProjects().contains(getPayload().getPathWithNamespace());
+        return navigator instanceof GitLabSCMNavigator && isMatch((GitLabSCMNavigator) navigator);
+    }
+
+    private boolean isMatch(@Nonnull GitLabSCMNavigator navigator) {
```

Comment:
@stephenc I know you are a busy man 😓 But do you have a chance to look at this. 🙏